doc/source/reference/c-api.ufunc.rst

```diff
@@ -93,12 +93,22 @@ Functions
         the corresponding 1-d loop function in the func array.
 
     :param types:
-        Must be of length (*nin* + *nout*) \* *ntypes*, and it
-        contains the data-types (built-in only) that the corresponding
-        function in the *func* array can deal with.
+
+        Char of length ``(nin + nout) * ntypes`` It encodes the
+        :ref:`PyArray_Descr.type_num` (built-in only) that the corresponding
+        function in the ``func`` array accepts. For a ufunc with two
+        ``ntypes``, one ``nin`` and one ``nout`` where the first function
+        accepts and returns ``int32`` and the second accepts and returns
+        ``int64``, ``types`` would be ``(char[]) {5, 5, 7, 7}`` since
```

Comment:
With the 2d notation above, this becomes `(char[2][2]) {{5, 5}, {7, 7}};`. To break symmetry between the 2s, I'd be inclined to use a binary op as an example.

Reply:
Yes, I'll use a comparison as an example instead.